### PR TITLE
DDP-5573 3/3: handle parent/child instance status updates

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

This PR implements the last requirement for DDP-5573. I think these are significant changes so putting these changes in its own PR for more focused review. Highlights of changes:

* Updated patch/put routes to handle updating parent instance's status.
* Updated patch/put routes to check parent instance read-only status before proceeding.
* Made some small optimizations and refactors.

Part 1: #644 
Part 2: #641 

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

- [ ] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [ ] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!

### CODE REVIEW

File: pepper-apis/docs/specification/src/components/responses.yml [hunk 1]
```diff
@@ -58,7 +58,10 @@ ActivityAnswersPatchBadPayloadResponse:
       schema:
         $ref: '../pepper.yml#/components/schemas/Error.ActivityAnswersPatchBadPayload'
 ActivityAnswersPatchInvalidResponse:
-  description: activity instance or question is read-only or answer submission failed validation
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/components/responses.yml:61]: @bskinner I updated the documentation descriptions to better spell out parent/child behavior.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java [hunk 2]
```diff
@@ -152,51 +149,63 @@ public Object handle(Request request, Response response) {
...
-            LanguageDto preferredUserLanguage = RouteUtil.getUserLanguage(request);
-            String isoLanguageCode = preferredUserLanguage.getIsoCode();
-            Long languageCodeId = preferredUserLanguage.getId();
+            LanguageDto preferredUserLangDto = RouteUtil.getUserLanguage(request);
+
+            User operatorUser = instanceSummary.getParticipantUser();
+            if (!participantGuid.equals(operatorGuid)) {
+                operatorUser = handle.attach(UserDao.class).findUserByGuid(operatorGuid)
+                        .orElseThrow(() -> new DDPException("Could not find operator with guid " + operatorGuid));
+            }
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java:202]: Small optimization: operator is almost always the participant, so we start with that and only do a query if operator is different. Also putting this down here so we only do this query if it passed everything else above.

**Comment by MocanaAtBroad** [comment]
Are there changes in this PR that were not in the two PRs that are merged into this one?

**Comment by yufengwng** [comment]
@MocanaAtBroad I've merged in the other two PRs and rebased this PR, so it should properly reflect the commits and "Files changed".

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**